### ISSUE TITLE
STIM 014 not needed for KIT data

### ISSUE THREAD

**Issue by monkeyman192** [mne-tools/mne-bids#51]
Ok, last issue for now I think...
I understand that the STIM 014 channel is created automatically for KIT data for the purpose of triggers, however I don't think it should be included in the channel list in the bids format as it isn't an actual channel.
Even if I specify the information to get triggers to work correctly and produce a correct looking event.tsv, the STIM 014 channel is still there.
I am thinking we could just check if the data is KIT data and exclude the STIM 014 channel from the counting/channel list? Or even just check if a STIM 014 channel exists and ignore it if it does as I assume it will never be a channel we want to consider.

**Comment by jasmainak** [comment]
> I understand that the STI 014 channel is created automatically for KIT data for the purpose of triggers

can you elaborate on this? I have not worked with KIT systems. There isn't a trigger channel for KIT systems? Is it a channel that MNE creates when you try to read the data?

**Comment by monkeyman192** [comment]
Yep, MNE creates the channel artificially for trigger data. You can set any of the misc channels as triggers, but the actual .con file doesn't contain any information to indicate that they are trigger channels.
The channel is created when the raw is, not when the actual data is read.

**Comment by jasmainak** [comment]
hmm ... well, in MNE-BIDS, the files saved are in *.fif format not *.con. So, technically it's not an issue because the sidecar should correspond to the file saved in the BIDS folder, not the original file

**Comment by teonbrooks** [comment]
with MNE-BIDS, we do perserve the original format. Are you saying that the synthetic channel appears in the Channel List? If so, then that is a problem given that the folder will contain the original files and that channel won't correspond to anything. If you can confirm that is the case, we can start compiling a list of issues, such as this one perhaps, that Mainak and I can work on at neuroinformatics

**Comment by teonbrooks** [comment]
i just realized that I kinda made it seem like it was just the two of us who could do it. i just meant that we will be sprinting there and might have a chance to work on it together. 

@monkeyman192, definitely feel free to file a PR fix for this if you would like. these days, I only get bouts of time for coding (e.g. neuroinformatics), but I have more time for review :)

**Comment by monkeyman192** [comment]
> Are you saying that the synthetic channel appears in the Channel List?

Yes, this is exactly what is happening. Have a look at the channels.tsv I have uploaded to see what the output looks like (you can also see what it looks like with the channel type fix I made...)
[sub-2630_ses-1_task-a_acq-a1_channels.txt](https://github.com/mne-tools/mne-bids/files/2199608/sub-2630_ses-1_task-a_acq-a1_channels.txt)

I am actually not sure what the best way to fix this would be... I had a bit of a go at trying to do what I said (detect a STI 014 channel and ignore it), but it is so hacky...

**Comment by monkeyman192** [comment]
ok, I think I have found a reasonably simple solution involving simply removing the channel entry from the dataframe before it is written to a file. will try do a PR at some point with the fix.

**Comment by monkeyman192** [comment]
ok, once I have gotten the current PR fixed I'll do a few small simple ones with fixes for a few of these issues.

**Comment by sappelhoff** [comment]
This is not only relevant for KIT data but also in the near future, when we are dealing with EEG data. For the EEG data it would be good if no STI014 would be added to `channels.tsv` by default.

BTW: Why is it called STI014? ... STI=Stimulus, okay. But 014?

**Comment by agramfort** [comment]
it's an historical name for old neuromag MEG vectorview systems...

>

**Comment by jasmainak** [comment]
as I said before, PR is welcome :-)

**Comment by monkeyman192** [comment]
I'll have a look at my fix when I get to work and see if it is easy enough to make as a standalone PR. I think it should be
